### PR TITLE
Added support for Wifi 5GHz (802.11a).

### PR DESCRIPTION
**Pull request by ftoure67**
Signed-off-by: ftoure67 <famory.toure@gmail.com>

Added support for Wifi 802.11a (5 GHz)

Related Issue: This PR fixes/closes {2990, 2914}

Replaced the Wifi channel grid by a combo list of channels. The code is still present but the grid is hidden.
Added the display of the country code.

Screenshots: https://photos.google.com/search/_tra_/photo/AF1QipMJpEdDwJjAGeHWhWkzQKhKRCJoh1ppSDObGawv

### CODE REVIEW

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiChannel.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package org.eclipse.kura.net.wifi;
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiChannel.java:14]: A copyright header is required for all the code contributions

File: kura/org.eclipse.kura.core.net/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -15,7 +15,7 @@ Import-Package: org.eclipse.kura;version="[1.0,2.0)",
  org.eclipse.kura.net.dhcp;version="[1.0,2.0)",
  org.eclipse.kura.net.firewall;version="[1.0,2.0)",
  org.eclipse.kura.net.modem;version="[2.1,2.2)",
- org.eclipse.kura.net.wifi;version="[2.0,2.1)",
+ org.eclipse.kura.net.wifi;version="[2.0.0,2.2.0)",
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.net/META-INF/MANIFEST.MF]: The format that we usually adopt is [2.0,2.1) without the bug fix version.
Since in this case the core.net bundle implements the net.api the range is restricted to a specific version, that is the reason it was [2.0,2.1) meaning that implements version 2.0 of the APIs. If in this case the API version has been pushed to 2.1 than the version should change to [2.1,2.2)

File: kura/org.eclipse.kura.linux.net/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -16,6 +16,7 @@ Import-Package: javax.microedition.io,
  org.apache.commons.net.util;version="3.1.0",
  org.eclipse.kura;version="[1.1,2.0)",
  org.eclipse.kura.comm;version="[1.0,2.0)",
+ org.eclipse.kura.command;version="1.2.0",
```
Comment (MMaiero) [inline kura/org.eclipse.kura.linux.net/META-INF/MANIFEST.MF]: If it is importing, it should use a version range.

File: kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/util/IwListChannelTool.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
+package org.eclipse.kura.linux.net.util;
```
Comment (MMaiero) [inline kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/util/IwListChannelTool.java:13]: A copyright header is required for all the code contributions

File: kura/org.eclipse.kura.net.admin/META-INF/MANIFEST.MF [hunk 5]
```diff
@@ -14,7 +14,7 @@ Import-Package: javax.microedition.io,
  org.eclipse.kura.core.configuration.metatype;version="[1.0,2.0)",
  org.eclipse.kura.core.linux.executor;version="[1.0,2.0)",
  org.eclipse.kura.core.linux.util;version="[1.0,2.0)",
- org.eclipse.kura.core.net;version="[1.0,2.0)",
+ org.eclipse.kura.core.net;version="[1.0,2.1)",
```
Comment (MMaiero) [inline kura/org.eclipse.kura.net.admin/META-INF/MANIFEST.MF:17]: This is a strange change.

File: kura/org.eclipse.kura.web2/META-INF/MANIFEST.MF [hunk 6]
```diff
@@ -88,5 +88,6 @@ Import-Package: com.eclipsesource.json;version="0.9.5",
  org.slf4j;version="1.6.4",
  org.w3c.dom,
  org.xml.sax
-Require-Bundle: com.gwt.user;bundle-version="[1.1.0,2.0.0)"
+Require-Bundle: com.gwt.user;bundle-version="[1.1.0,2.0.0)",
+ org.eclipse.kura.api
```
Comment (MMaiero) [inline kura/org.eclipse.kura.web2/META-INF/MANIFEST.MF]: Why this change?

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/GwtWifiChannelFrequency.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
+package org.eclipse.kura.web.shared.model;
```
Comment (MMaiero) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/GwtWifiChannelFrequency.java:13]: A copyright header is required for all the code contributions

**Review by MMaiero — COMMENTED** [review]
Thanks for the contribution. As you can see from the CI build results, the new code needs to be associated with corresponding tests. Please also review the comments made in the code

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiChannel.java (hunk 1 above)
Comment (ftoure67) [reply kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiChannel.java:14]: OK.

File: kura/org.eclipse.kura.core.net/META-INF/MANIFEST.MF (hunk 2 above)
Comment (ftoure67) [reply kura/org.eclipse.kura.core.net/META-INF/MANIFEST.MF]: OK. I will update it.

File: kura/org.eclipse.kura.linux.net/META-INF/MANIFEST.MF (hunk 3 above)
Comment (ftoure67) [reply kura/org.eclipse.kura.linux.net/META-INF/MANIFEST.MF]: Not needed. I will remove it.

File: kura/org.eclipse.kura.net.admin/META-INF/MANIFEST.MF (hunk 5 above)
Comment (ftoure67) [reply kura/org.eclipse.kura.net.admin/META-INF/MANIFEST.MF:17]: Not needed. I will remove it.

File: kura/org.eclipse.kura.web2/META-INF/MANIFEST.MF (hunk 6 above)
Comment (ftoure67) [reply kura/org.eclipse.kura.web2/META-INF/MANIFEST.MF]: Not needed. I will remove it.

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/GwtWifiChannelFrequency.java (hunk 7 above)
Comment (ftoure67) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/GwtWifiChannelFrequency.java:13]: OK.

**Comment by MMaiero** [comment]
@ftoure67 Can you please check your code? Because the CI fails building the Web UI: https://ci.eclipse.org/kura/job/multibranch/view/change-requests/job/PR-3303/5/consoleFull

**Comment by ftoure67** [comment]
Hi Matteo,

My bad. Two methods were duplicated during the merge.
I have pushed again.
Once the CI is passed, please also check if the new Wifi interface is OK
for you since it's a big change compared to the previous version.

Regards,

Famory TOURE

Le jeu. 4 mars 2021 à 09:19, Matteo Maiero <notifications@github.com> a
écrit :

**Comment by MMaiero** [comment]
Hi @ftoure67 
what device did you use as a reference?
I have just tried with one of ours (without additional changes from your PR) and this is what I see from the UI:

Screenshot 2021-03-08 at 08 58 06

**Comment by ftoure67** [comment]
Hi Matteo,

I'm using a SolidSense gateway with Raspberry-Pi 2/3 load installed. The
command executed is "iwlist wlan0 channel" where wlan0 is the name of the
Wifi interface. The result of the command seems to be empty in your case.
I can run the test on a Raspberry-Pi 3 or 4 to double check.

Regards,

Famory TOURE

Le lun. 8 mars 2021 à 08:59, Matteo Maiero <notifications@github.com> a
écrit :

**Comment by MMaiero** [comment]
Be aware that iwlist is deprecated in many systems.
For example, in my system I have no iwlist available.
Maybe let's try to use iw or iw dev to be compliant with the latest suggested binaries for wifi

**Comment by MMaiero** [comment]
As agreed with @ftoure67 this PR is superseded by #3329
